lib/model/folder_sendrecv.go

```diff
@@ -497,21 +505,22 @@ func (f *sendReceiveFolder) processDeletions(fileDeletions map[string]protocol.F
 		}
 
 		l.Debugln(f, "Deleting file", file.Name)
+		f.resetPullError(file.Name)
 		if update, err := f.deleteFile(file, scanChan); err != nil {
 			f.newPullError(file.Name, errors.Wrap(err, "delete file"))
 		} else {
 			dbUpdateChan <- update
 		}
 	}
 
-	for i := range dirDeletions {
+	for _, dir := range dirDeletions {
 		select {
 		case <-f.ctx.Done():
 			return
 		default:
 		}
 
-		dir := dirDeletions[len(dirDeletions)-i-1]
+		f.resetPullError(dir.Name)
```

Comment:
Now we try to delete `a\b` before `a\b\c`. Seems wrong?

Reply:
Thanks!  
I think I should take a break after fixing the two known messes I already caused. I am producing so much BS at the moment.

Reply:
Ah, I missed this behavior change in the review as well :/